### PR TITLE
Integration tests along with multiple iOS versions coverage

### PR DESCRIPTION
**Pull request by huylenq**
Main changes are:

- Added `IntegrationTests.swift` along with the adhoc API client `TDAPI.swift`
- Reconfiguration of CircleCI `~/.circleci/config.yml` along with `scripts/ci_test` and `scripts/ci_pod_install`

The PR's diffs have a lot of noises due to the new `TestHost` app target (a necessity to able to test for app lifecycle events).

### CODE REVIEW

File: TreasureDataTests/IntegrationTests.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    static var apiKey: String!
+
+    static let TargetDatabase = "ios_it"
+    static var sessionPrefix = uuid()
+
+    /** Tables used in a single test */
+    var tempTables: [String] = []
+
+    override class func setUp() {
+        apiKey = ProcessInfo.processInfo.environment["TD_DEV_MASTER_KEY"]
```
Comment (huylenq) [inline TreasureDataTests/IntegrationTests.swift]: I probably will change this to target and use a production key instead

**Comment by huylenq** [comment]
CircleCI appears could be flaky on it's macos environment or XCode, [this build](https://circleci.com/gh/treasure-data/td-ios-sdk/54) failed with:

```
#!/bin/bash -eo pipefail
./scripts/ci_test 'platform=iOS Simulator,name=iPhone 7,OS=12.1' logs/xcbuild_ios_12.log test_results/ios_12.xml
^D^Dxcodebuild: error: Unable to find a destination matching the provided destination specifier:
		{ platform:iOS Simulator, OS:12.1, name:iPhone 7 }

	The requested device could not be found because no available devices matched the request.

	Ineligible destinations for the "TreasureData" scheme:
		{ platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Generic iOS Device }
		{ platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Generic iOS Simulator Device }
Exited with code 70
```
And succeed after [a rebuild](https://circleci.com/gh/treasure-data/td-ios-sdk/55), This seldom happens but I encountered it once or twice, probably need to address if it occurs more often in the future.

File: TestHost/ViewController.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//
+//  ViewController.swift
+//  TestHost
+//
+//  Created by Huy TD on 1/28/19.
+//  Copyright © 2019 Huy TD. All rights reserved.
```
Comment (vinhvd) [inline TestHost/ViewController.swift]: Look like it make sense to change the Copyright to Arm Treasure Data ?

File: TreasureDataTests/IntegrationTests.swift [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            database: IntegrationTests.TargetDatabase)
+        for event in eventRecords {
+            if (event[1] as! String == "in_session_event") {
+                XCTAssertEqual(event[0] as? String, sessionId)
+            } else if (event[1] as! String == "not_in_session_event") {
+                XCTAssertEqual(event[0] as! String, "")
+            }
+        }
+    }
+
```
Comment (vinhvd) [inline TreasureDataTests/IntegrationTests.swift]: Is it able to add a test case for reset/forgetGUID ?

File: TreasureDataTests/IntegrationTests.swift [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    static let TargetDatabase = "ios_it"
+    static var sessionPrefix = uuid()
+
+    /** Tables used in a single test */
+    var tempTables: [String] = []
+
+    override class func setUp() {
+        apiKey = ProcessInfo.processInfo.environment["TD_DEV_MASTER_KEY"]
+        api = TDAPI.init(endpoint: "https://api-development.treasuredata.com", apiKey: apiKey)
```
Comment (kietdo360) [inline TreasureDataTests/IntegrationTests.swift]: Let put both the endpoint and API key to environment variable

File: TreasureDataTests/IntegrationTests.swift [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        sdkClient = nil
+    }
+
+    func testCustomEvent() {
+        sdkClient.enableCustomEvent()
+        let table = newTempTable()
+        let event = sdkClient.addEvent(["message": "Guten Morgen!"], table: table)
+        sdkClient.uploadEvents()
+        if event != nil {
+            let result = try! IntegrationTests.api.stubbornQuery("select message from \(table) limit 1", database: IntegrationTests.TargetDatabase)
```
Comment (kietdo360) [inline TreasureDataTests/IntegrationTests.swift]: How long to loop before receiving the result ?

Comment (huylenq) [reply TreasureDataTests/IntegrationTests.swift]: I observed it could span from somewhere from 100 to 300 secs, usually it around ~200s.

**Comment by huylenq** [comment]
Hmm, something wrongs on CI, it constantly failed with "2019-03-25 21:53:11.067 xcodebuild[741:5293] Error Domain=IDETestOperationsObserverErrorDomain Code=7 "Exceeded max restart count of 2" UserInfo={NSLocalizedDescription=Exceeded max restart count of 2}
"

I tried to rerun the previous succeed job on 3a813e6, it still failed with the same message.

**Pull request merged**